library/hdfs-library/src/main/java/uk/gov/gchq/gaffer/hdfs/operation/MapReduce.java

```diff
@@ -39,23 +38,25 @@
  * @see Builder
  */
 public interface MapReduce {
-    List<String> getInputPaths();
+    Map<String, String> getInputMapperPairs();
 
-    void setInputPaths(final List<String> inputPaths);
+    void setInputMapperPairs(final Map<String, String> inputMapperPairs);
 
-    default void addInputPaths(final List<String> inputPaths) {
-        if (null == getInputPaths()) {
-            setInputPaths(new ArrayList<>(inputPaths));
+    default void addInputMapperPairs(final Map<String, String> inputMapperPairs) {
+        if (null == getInputMapperPairs()) {
+            setInputMapperPairs(inputMapperPairs);
         } else {
-            getInputPaths().addAll(inputPaths);
+            getInputMapperPairs().putAll(inputMapperPairs);
         }
     }
 
-    default void addInputPath(final String inputPath) {
-        if (null == getInputPaths()) {
-            setInputPaths(Lists.newArrayList(inputPath));
+    default void addInputMapperPair(final String inputPath, final String mapperGeneratorClassName) {
+        Map<String, String> inputMapperMap = new HashMap<>();
```

Comment:
Could these two lines be moved inside the first branch of the if statement below? This object is only used if the condition is true.